tests/app/broadcast_message/test_rest.py

```diff
@@ -178,23 +178,106 @@ def test_create_broadcast_message_400s_if_json_schema_fails_validation(
+            'content': 'Some tailor made broadcast content',
+            'service_id': str(sample_broadcast_service.id),
+            'created_by': str(sample_broadcast_service.created_by_id),
+        },
+        service_id=sample_broadcast_service.id,
+        _expected_status=400
+    )
+
+    assert len(response['errors']) == 1
+    assert response['errors'][0]['error'] == 'ValidationError'
+    # The error message for oneOf is ugly, non-deterministic in ordering
+    # and contains some UUID, so let’s just pick out the important bits
+    assert (
+        ' is valid under each of '
+    ) in response['errors'][0]['message']
+    assert (
+        '{required: [content]}'
+    ) in response['errors'][0]['message']
+    assert (
+        '{required: [template_id]}'
+    ) in response['errors'][0]['message']
+
+
+def test_create_broadcast_message_400s_if_reference_and_template_provided(
+    admin_request,
+    sample_broadcast_service,
+):
+    template = create_template(sample_broadcast_service, BROADCAST_TYPE)
+    response = admin_request.post(
+        'broadcast_message.create_broadcast_message',
+        _data={
+            'template_id': str(template.id),
+            'reference': 'abc123',
             'service_id': str(sample_broadcast_service.id),
             'created_by': str(sample_broadcast_service.created_by_id),
         },
         service_id=sample_broadcast_service.id,
         _expected_status=400
     )
-    assert response['errors'] ==[
-        {'error': 'ValidationError', 'message': 'template_id is not a valid UUID'},
-        {'error': 'ValidationError', 'message': 'Additional properties are not allowed (content was unexpected)'},
-    ]
+
+    assert len(response['errors']) == 1
+    assert response['errors'][0]['error'] == 'ValidationError'
+    # The error message for oneOf is ugly, non-deterministic in ordering
+    # and contains some UUID, so let’s just pick out the important bits
+    assert (
+        ' is valid under each of '
+    ) in response['errors'][0]['message']
+    assert (
+        '{required: [reference]}'
+    ) in response['errors'][0]['message']
+    assert (
+        '{required: [template_id]}'
+    ) in response['errors'][0]['message']
+
+
+def test_create_broadcast_message_400s_if_reference_not_provided_with_content(
+    admin_request,
+    sample_broadcast_service,
+):
+    response = admin_request.post(
+        'broadcast_message.create_broadcast_message',
+        _data={
+            'content': 'Some tailor made broadcast content',
+            'service_id': str(sample_broadcast_service.id),
+            'created_by': str(sample_broadcast_service.created_by_id),
+        },
+        service_id=sample_broadcast_service.id,
+        _expected_status=400
+    )
+    assert len(response['errors']) == 1
+    assert response['errors'][0]['error'] == 'ValidationError'
+    assert response['errors'][0]['message'].endswith(
+        'is not valid under any of the given schemas'
+    )
 
```

Comment:
Should we also have a test for when neither template_id nor content is provided?